### ISSUE TITLE
Broken URI leads to "The BOM does not conform to the CycloneDX BOM standard as defined by the XSD"

### ISSUE THREAD

**Issue by mfriedenhagen** [CycloneDX/cyclonedx-maven-plugin#46]
Hello,

another error from my synthetic mega-pom :-) mentioned in #45:  
```
org.apache.maven.plugin.MojoExecutionException: The BOM does not conform to the CycloneDX BOM standard as defined by the XSD
    at org.cyclonedx.maven.BaseCycloneDxMojo.execute (BaseCycloneDxMojo.java:608)
```

After setting a breakpoint in /Users/user/.m2/repository/org/cyclonedx/cyclonedx-core-java/2.6.3/cyclonedx-core-java-2.6.3-sources.jar!/org/cyclonedx/BomParser.java:213
I copied the following exceptions, I think all of which are provoked by one wrong URL :

```
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27009; columnNumber: 81; cvc-datatype-valid.1.2.1: 'http://www.mail-archive.com/dev%karaf.apache.org/' is not a valid value for 'anyURI'.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27009; columnNumber: 81; cvc-type.3.1.3: The value 'http://www.mail-archive.com/dev%karaf.apache.org/' of element 'url' is not valid.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27127; columnNumber: 81; cvc-datatype-valid.1.2.1: 'http://www.mail-archive.com/dev%karaf.apache.org/' is not a valid value for 'anyURI'.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27127; columnNumber: 81; cvc-type.3.1.3: The value 'http://www.mail-archive.com/dev%karaf.apache.org/' of element 'url' is not valid.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27167; columnNumber: 81; cvc-datatype-valid.1.2.1: 'http://www.mail-archive.com/dev%karaf.apache.org/' is not a valid value for 'anyURI'.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27167; columnNumber: 81; cvc-type.3.1.3: The value 'http://www.mail-archive.com/dev%karaf.apache.org/' of element 'url' is not valid.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27207; columnNumber: 81; cvc-datatype-valid.1.2.1: 'http://www.mail-archive.com/dev%karaf.apache.org/' is not a valid value for 'anyURI'.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27207; columnNumber: 81; cvc-type.3.1.3: The value 'http://www.mail-archive.com/dev%karaf.apache.org/' of element 'url' is not valid.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27247; columnNumber: 81; cvc-datatype-valid.1.2.1: 'http://www.mail-archive.com/dev%karaf.apache.org/' is not a valid value for 'anyURI'.
org.xml.sax.SAXParseException; systemId: file:/Users/user/ws/git.example.com/mamido/mamido-pom/target/it/mamido-pom-test/target/bom.xml; lineNumber: 27247; columnNumber: 81; cvc-type.3.1.3: The value 'http://www.mail-archive.com/dev%karaf.apache.org/' of element 'url' is not valid.
```

It boils down to one of the parents:
https://repo1.maven.org/maven2/org/apache/karaf/karaf/4.2.6/karaf-4.2.6.pom
As you can see, the percent encoding of "Karaf Dev" mailing list is incomplete.

**Comment by stevespringett** [comment]
Thanks again for reporting the issue. I was able to replicate using a pom with a single dependency of:
```xml
<groupId>org.apache.karaf</groupId>
<artifactId>org.apache.karaf.util</artifactId>
<version>4.2.6</version>
```

The issue is interesting. On the surface, it seems to be a difference between what the Java URL class is a valid URL and what the W3C defines as a valid URL. The easy path would be to simply wrap it into a CNAME, but I'd rather attempt to determine root cause and address it that way instead.

**Comment by mfriedenhagen** [comment]
I think percent signs are only valid in URLs when used with URL encoding and so the URL is definitely broken. I would set it empty, placing it in CDATA does not help here IMO.

**Comment by stevespringett** [comment]
Fixed in v1.6.3.
